core/core-spacemacs-buffer.el

```diff
@@ -841,12 +845,20 @@ already exist, and switch to it."
   (switch-to-buffer spacemacs-buffer-name)
   (spacemacs//redisplay))
 
+(add-hook 'window-size-change-functions 'spacemacs-buffer//resize-on-hook)
+
+(defun spacemacs-buffer//resize-on-hook (frame)
+  (let ((space-win (get-buffer-window spacemacs-buffer-name))
+        (frame-win (frame-selected-window frame)))
+    (when (and space-win
+               (not (window-minibuffer-p frame-win)))
+      (save-selected-window
+        (select-window space-win)
```

Comment:
I agree, I tried that first and it didn't work, but I think that was likely because I was still killing the spacemacs-buffer instead of erasing it.